internal/globalaccounts/globalaccounts.go

```diff
@@ -0,0 +1,177 @@
+package globalaccounts
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/kyma-project/kyma-environment-broker/internal/events"
+	"github.com/kyma-project/kyma-environment-broker/internal/k8s"
+	"github.com/kyma-project/kyma-environment-broker/internal/storage"
+	"github.com/kyma-project/kyma-environment-broker/internal/storage/dbmodel"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2/clientcredentials"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	k8scfg "sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+type result struct {
+	GlobalAccountGUID string `json:"globalAccountGUID"`
+}
+
+type svcConfig struct {
+	ClientID       string
+	ClientSecret   string
+	AuthURL        string
+	SubaccountsURL string
+}
+
+func Run(c Config) {
+	ctx := context.Background()
+	logs := logrus.New()
+	logs.Infof("*** Start at: %s ***", time.Now().Format(time.RFC3339))
+
+	svcConfig := svcConfig{
+		ClientID:     c.AccountServiceID,
+		ClientSecret: c.AccountServiceSecret,
+		AuthURL:      c.AccountServiceURL,
+	}
+
+	oauthConfig := clientcredentials.Config{
+		ClientID:     svcConfig.ClientID,
+		ClientSecret: svcConfig.ClientSecret,
+		TokenURL:     svcConfig.AuthURL,
+	}
+
+	db, connection, err := storage.NewFromConfig(
+		c.Database,
+		events.Config{},
+		storage.NewEncrypter(c.Database.SecretKey),
+		logs.WithField("service", "storage"))
+	fatalOnError(err, logs)
+
+	defer func() {
+		err = connection.Close()
+		if err != nil {
+			logs.Error(err)
+		}
+	}()
+
+	kcpK8sClient, err := getKcpClient()
+	fatalOnError(err, logs)
+
+	gvk, err := k8s.GvkByName(k8s.KymaCr)
+	fatalOnError(err, logs)
+
+	svc := oauthConfig.Client(ctx)
+
+	kymas := unstructured.UnstructuredList{}
+	kymas.SetGroupVersionKind(gvk)
+	err = kcpK8sClient.List(ctx, &kymas)
+	fatalOnError(err, logs)
+
+	var data strings.Builder
+	for _, kyma := range kymas.Items {
+		runtimeId := kyma.GetName() // name of kyma is runtime id
+		runtimeIDFilter := dbmodel.InstanceFilter{RuntimeIDs: []string{runtimeId}}
+
+		instances, _, _, err := db.Instances().List(runtimeIDFilter)
+		if err != nil {
+			logs.Error(err)
+			continue
+		}
+		if len(instances) == 0 {
+			logs.Errorf("no instance for runtime id %s", runtimeId)
+			continue
+		}
+		if len(instances) > 1 {
+			logs.Errorf("more than one instance for runtime id %s", runtimeId)
+			continue
+		}
+		instance := instances[0]
+		if instance.SubAccountID == "" {
+			logs.Errorf("instance have empty SA %s", instance.SubAccountID)
+			continue
+		}
+		if instance.GlobalAccountID == "" {
+			logs.Errorf("instance have empty GA %s", instance.GlobalAccountID)
+			continue
+		}
+
+		request, err := http.NewRequest(http.MethodGet, fmt.Sprintf(c.AccountServiceURL, instance.SubAccountID), nil)
```

Comment:
You could extract this part to separate function (returning response, error tuple).